skynet/agent.py

```diff
@@ -41,3 +49,75 @@ async def create_tcpserver(port):
         await tcpserver.serve_forever()
     except KeyboardInterrupt:
         tcpserver.close()
+
+
+# Endpoints for the autoscaler to query the current state of the system
+class StateResponse(BaseModel):
+    request_status: str
+    current_state: str
+    desired_state: str
+
+
+class CurrentStateResponse(BaseModel):
+    current_state: str
+    active_connections: int
+    stress_level: float
+
+
+async def get_current_state():
```

Comment:
This function blocks the asyncio loop because you are using standard library funcions, which block. You need to use aiofiles for asyncio friendly filesystem APIs.

More importantly, why do we need to read the file every time it's queried? What would cause its value to change? Is anyone else writing to that file?